### DISCUSSION TITLE
Simple chat demo exception when SIGNOUT is selected

### DISCUSSION THREAD

**Discussion by Autoneer** [SchneiderInfosystems/FB4D#131] [Q&A] [answered]
When I run the chat sample and SIGNOUT I get an http exception

First chance exception at $00007FFBD558478C. Exception class ENetHTTPResponseException with message 'Error reading data: (12017) The operation has been canceled'. Process FSSimpleChat.exe (7988)

is this a timing issue?

**Comment by SchneiderInfosystems** [comment, marked as answer]
No, it is not a timing issue. This is an expected "silent" exception generated by the underlying HTTP component when the Firestore listener is closed.  

You will see this exception only in the debugger but not in the running application.

**Reply by Autoneer** [reply]
I also have the following issue with SIGNOUT.
On windows it works 100%
On Android, the app freezes and I need to kill it on the phone and restart the app to be able to sign in with a different user.?
any idea why this would happen?

**Reply by SchneiderInfosystems** [reply]
I cannot reproduce this problem. I have the application FSSimpleChat running on my Android and I can log out and log in with another user without any problems. 
Can you test this demo app on your Android? If it should run, check the differences between your solution and my FSSimpleChat.

**Reply by Autoneer** [reply]
It is the SimpleChat demo I am talking about

**Reply by SchneiderInfosystems** [reply]
Can you make a video to show me the issue on your Android device? Does the issue also appear if you run the app without an attached debugger? As explained I cannot reproduce it with my Android Device so I need more detailed information to help. Android version? Device? Can you check the cat log when freezing the app?
